app/src/main/java/xyz/whereuat/whereuat/ui/adapters/KeyLocationAdapter.java

```diff
@@ -0,0 +1,94 @@
+    private static final int[] BOUND_VIEWS = new int[] {R.id.key_loc_item};
+    private static AppCompatActivity mActivity;
+
+    public KeyLocationAdapter(AppCompatActivity activity) {
+        super(activity, R.layout.key_loc_list_item, null, BOUND_COLS, BOUND_VIEWS, 0);
+        mActivity = activity;
+    }
+
+    @Override
+    public void bindView (View view, final Context context, final Cursor cursor) {
+        String name = cursor.getString(cursor.getColumnIndex(KeyLocationEntry.COLUMN_NAME));
+        float lat = cursor.getFloat(cursor.getColumnIndex(KeyLocationEntry.COLUMN_LATITUDE));
+        float lng = cursor.getFloat(cursor.getColumnIndex(KeyLocationEntry.COLUMN_LONGITUDE));
+        String coords = String.format("%f, %f", lat, lng);
+
+        ((LatoTextView) view.findViewById(R.id.key_loc_name)).setText(name);
+        ((LatoTextView) view.findViewById(R.id.key_loc_coords)).setText(coords);
+        final int id = cursor.getInt(cursor.getColumnIndex(KeyLocationEntry._ID));
+
+        // Set up the click listeners for the buttons.
+        (view.findViewById(R.id.edit_btn)).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                handleWithEditKeyLocDialog(id);
+            }
+        });
+        (view.findViewById(R.id.delete_btn)).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                handleDeleteKeyLoc(context, id);
+            }
+        });
+    }
+
+    private void handleDeleteKeyLoc(final Context context, final int id) {
+        Toast.makeText(mContext, "Deleting key location...", Toast.LENGTH_SHORT).show();
+        AsyncExecutor.service.submit(new Runnable() {
+            @Override
+            public void run() {
+                final int deleted = KeyLocationUtils.buildDeleteByIdCommand(context, id).call();
+                mActivity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (deleted == 1) {
+                            KeyLocationActivity.notifyOfKeyLocationChange(mActivity);
+                            Toast.makeText(mActivity, "Key location deleted.", Toast.LENGTH_SHORT)
+                                    .show();
+                        } else {
+                            Toast.makeText(mActivity, "Key location couldn't be deleted :(",
+                                    Toast.LENGTH_SHORT).show();
+                        }
+                    }
+                });
+            }
+        });
+    }
+
+    private void handleWithEditKeyLocDialog(int id) {
+        FragmentManager fm = mActivity.getFragmentManager();
+        EditKeyLocationFragment add_contact_dialog = new EditKeyLocationFragment();
+        Bundle args = new Bundle();
+        args.putInt(Constants.KEY_LOC_ID_BUNDLE_EXTRA, id);
+        add_contact_dialog.setArguments(args);
```

Comment:
Could do the 88-91 functionality inside of a builder method. Did something similar with the key location confirmation dialog fragment.